packages/shared/src/components/HorizontalScroll/useScrollManagement.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { RefObject, useCallback, useEffect, useState } from 'react';
+import useDebounceFn from '../../hooks/useDebounceFn';
+
+interface ScrollManagementReturn {
+  isAtStart: boolean;
+  isAtEnd: boolean;
+}
+export const useScrollManagement = (
+  ref: React.RefObject<HTMLElement>,
+  onScroll: (ref: RefObject<HTMLElement>) => void,
+): ScrollManagementReturn => {
+  const [isAtStart, setIsAtStart] = useState(true);
+  const [isAtEnd, setIsAtEnd] = useState(false);
+
+  const checkScrollPosition = useCallback(() => {
+    if (ref.current) {
+      onScroll?.(ref);
+      const { scrollLeft, scrollWidth, clientWidth } = ref.current;
+
+      setIsAtStart(scrollLeft === 0);
+      setIsAtEnd(scrollLeft + clientWidth >= scrollWidth);
+    }
+  }, [ref, onScroll]);
+
+  const [debouncedOnScroll] = useDebounceFn(checkScrollPosition, 100);
+
+  useEffect(() => {
```

Comment:
Probably yes, but this needs to be a effect?
(Eg we cant leverage useCallback as we technically only add event listener?)

Reply:
We can probably use useEventListener and I would use requestAnimationFrame instead (or in addition) to debounce to make it smooth and not impact scrolling.

Reply:
Addressed feedback as @capJavert suggested

Reply:
Based on Ante's comment

> hey all good, one thing I might have missed is that we don't really need requestAnimationFrame since you are not doing any layout manipulation directly, you are just reading values, so if not much of a hassle I think remove it to not complicate the code, sorry if going back and forth 

removing requestAnimationFrame additions

Reply:
Approved as well 🙌